snoop/utils.py

```diff
@@ -1,5 +1,64 @@
 import subprocess
 import re
+import exifread
+from datetime import datetime
+
+def extract_gps_location(tags):
+    def ratio_to_float(ratio):
+        return float(ratio.num) / ratio.den
+
+    def convert(value):
+        d = ratio_to_float(value.values[0])
+        m = ratio_to_float(value.values[1])
+        s = ratio_to_float(value.values[2])
+        return d + (m / 60.0) + (s / 3600.0)
+
+    tags = {key: tags[key] for key in tags.keys() if key.startswith('GPS')}
+
+    lat = tags.get('GPS GPSLatitude')
+    lat_ref = tags.get('GPS GPSLatitudeRef')
+    lon = tags.get('GPS GPSLongitude')
+    lon_ref = tags.get('GPS GPSLongitudeRef')
+
+    if not lat or not lon or not lat_ref or not lon_ref:
```

Comment:
How about `if not (land and lng and lat_ref and lng_ref)`? By the way, `0.0` is a valid value for lat or lng. So maybe: `if any(v is None for v in [lat, lat_ref, lng, lng_ref])`?